apm-agent-core/src/main/java/co/elastic/apm/agent/bci/classloading/IndyPluginClassLoader.java

```diff
@@ -68,7 +68,7 @@ private static ClassLoader getParent(@Nullable ClassLoader targetClassLoader, Cl
             // The list of packages not to load should correspond with matching dependency exclusions from the apm-agent-core in apm-agent-plugins/pom.xml
             // As we're using a custom logging facade, plugins don't need to refer to the agent-bundled log4j2 or slf4j.
             return new DiscriminatingMultiParentClassLoader(
-                agentClassLoader, not(startsWith("org.apache.logging.log4j").and(not(startsWith("org.slf4j")))),
+                agentClassLoader, not(startsWith("org.apache.logging.log4j").and(not(startsWith("org.slf4j")))).and(not(startsWith("co.elastic.logging.log4j2"))),
```

Comment:
`not(startsWith("org.apache.logging.log4j").and(not(startsWith("org.slf4j"))))` looks suspicious
 to me, I would have expected `not(startsWith("org.apache.logging.log4j")).and(not(startsWith("org.slf4j")))`.

Reply:
Yes, you rightfully think it is suspicious, but your proposal to exclude ECS logging is not something we want to do - we package ECS logging within the agent for the ECS-reformatting and we want to use our version rather than a different one provided as dependency.

We may exclude this package only for the `IndyPluginClassLoader` created for this specific plugin, but I have doubts about this plugin altogether. I think that going forward, the proper way to get ECS format when using the APM agent is through the ECS-reformating feature. There will still be a way to add it manually as a dependency for special cases, but in this case the service name can be manually configured. I am not too fond of this added complication coming from us instrumenting ECS logging.
@felixbarny WDYT?

Reply:
The apm-log-shadding-plugins use `PluginClassLoaderRootPackageCustomizer` to add `co.elastic.logging` to the list of plugin packages. So shouldn't the apm-log-shadding-plugins always load `co.elastic.logging.*` from the agent (with or without this change)?

Reply:
That's right! They are loaded from the plugin CL (not the agent CL). 
OK, then let's go with that, but please verify manually that the ECS-reformating works as expected with this change in place.

Reply:
Please confirm that everything works as expected ASAP, as we want to release 1.29.0 soon.

Reply:
And please fix the improper matcher chaining as well

Reply:
Is `not(startsWith("org.slf4j"))` still needed since you are using your own logging faced now?

Reply:
I tested `log_ecs_reformatting=SHADE`, `log_ecs_reformatting=REPLACE`, `log_ecs_reformatting=OVERRIDE`. In case of `SHADE` and `REPLACE` a *.ecs.json file was created contain the logs in ECS format. In case of `OVERRIDE` the original of file contains the logs in ECS format.

I also noticed the message `co.elastic.apm.agent.bci.bytebuddy.ErrorLoggingListener - co.elastic.logging.log4j2.EcsLayout$Builder refers to a missing class.` in the agent logs, which I fixed in [7f7ce59](https://github.com/elastic/apm-agent-java/pull/2458/commits/7f7ce5979d8ece4800a90efca76145b00f4e01b8)

Reply:
> Is not(startsWith("org.slf4j")) still needed since you are using your own logging faced now?

Yes, we need slf4j to satisfy stagemonitor's logging needs.

Reply:
Thanks for verifying